### ISSUE TITLE
macOS: book.save(path) clobbers the book object

### ISSUE THREAD

**Issue by wkschwartz** [xlwings/xlwings#1129]
On macOS, saving a `Book` object renders the object unusable: you can't use its `sheets` object, its `fullname` property, etc. After doing some digging with `pdb`, the line at fault appears to be https://github.com/xlwings/xlwings/blob/b75e15bdf7c4459d185ffc7e4a427be64ece7cf1/xlwings/_xlmac.py#L244 Details below the fold.

**Related issues**: #626.
**Versions**: macOS 10.13.6, Excel for Mac 2011 version 14.2.0 (120402), Python 3.7.3, xlwings 0.15.8, and appscripts 1.1.0

<details>

Once `repr(book1)` looks like `<Book [k.missing_value]>`, it's unusable. It doesn't matter if the save location is in the current working directory or somewhere else. (I include the `tempfile` experiment here because that's how I originally found the problem.)
```python-console
$ python
Python 3.7.3 (default, Mar 27 2019, 09:23:39) 
[Clang 10.0.0 (clang-1000.11.45.5)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> import tempfile; from pathlib import Path; import xlwings
>>> xlwings.__version__
'0.15.8'
>>> xlwings.apps
Apps([<Excel App 38595>])
>>> app = xlwings.apps[38595]; app
<Excel App 38595>
>>> book1 = app.books.add(); book1
<Book [Sheet4]>
>>> book2 = app.books.add(); book2
<Book [Sheet5]>
>>> tempdir = tempfile.TemporaryDirectory()
>>> temp_path = (Path(tempdir.name) / Path(book1.fullname).with_suffix('.xlsx')).resolve(); temp_path
PosixPath('/private/var/folders/6b/y6586bhs6ldfmxwrdj_c87l80000gn/T/tmp71p58nf0/Sheet4.xlsx')
>>> book1.save(str(temp_path))
>>> book1
<Book [k.missing_value]>
>>> book2
<Book [Sheet5]>
>>> book2.save('local.xlsx')
>>> book2
<Book [k.missing_value]>
>>> app.books
Books([<Book [Sheet4.xlsx]>, <Book [local.xlsx]>])
>>> app.books[0].fullname
'/private/var/folders/6b/y6586bhs6ldfmxwrdj_c87l80000gn/T/tmp71p58nf0/Sheet4.xlsx'
>>> app.books[1].fullname
'/path/to/cwd/local.xlsx'
>>> app.books[0].close()
>>> app.books
Books([<Book [local.xlsx]>])
>>> app.books[0].close()
>>> tempdir.cleanup()
>>> import os; os.unlink('local.xlsx')
```

</details>

**Comment by fzumstein** [comment]
I would say this is not just related to #626 but actually the same issue?

**Comment by fzumstein** [comment]
closed #1372
